airbyte-integrations/connectors/source-bing-ads/source_bing_ads/reports.py

```diff
@@ -0,0 +1,252 @@
+        request_kwargs = {
+            "service_name": None,
+            "account_id": account_id,
+            "operation_name": self.operation_name,
+            "is_report_service": True,
+            "params": {"download_parameters": ReportingDownloadParameters(**params)},
+        }
+        return self.client.request(**request_kwargs)
+
+    def get_report_request(
+        self,
+        account_id: str,
+        exclude_column_headers: bool,
+        exclude_report_footer: bool,
+        exclude_report_header: bool,
+        report_file_format: str,
+        return_only_complete_data: bool,
+        time: sudsobject.Object,
+    ) -> sudsobject.Object:
+        reporting_service = self.client.get_service(self.service_name)
+        report_request = reporting_service.factory.create(f"{self.report_name}Request")
+        if not self.aggregation_disabled:
+            report_request.Aggregation = self.client.report_aggregation
+
+        report_request.ExcludeColumnHeaders = exclude_column_headers
+        report_request.ExcludeReportFooter = exclude_report_footer
+        report_request.ExcludeReportHeader = exclude_report_header
+        report_request.Format = report_file_format
+        report_request.FormatVersion = "2.0"
+        report_request.ReturnOnlyCompleteData = return_only_complete_data
+        report_request.Time = time
+        report_request.ReportName = self.report_name
+        # Defines the set of accounts and campaigns to include in the report.
+        scope = reporting_service.factory.create("AccountThroughCampaignReportScope")
+        scope.AccountIds = {"long": [account_id]}
+        scope.Campaigns = None
+        report_request.Scope = scope
+
+        columns = reporting_service.factory.create(f"ArrayOf{self.report_name}Column")
+        getattr(columns, f"{self.report_name}Column").append(self.report_columns)
+        report_request.Columns = columns
+        return report_request
+
+    def parse_response(self, response: sudsobject.Object, **kwargs: Mapping[str, Any]) -> Iterable[Mapping]:
+        if response is not None:
+            for row in response.report_records:
+                yield {column: self.get_column_value(row, column) for column in self.report_columns}
+
+        yield from []
+
+    def get_column_value(self, row: _RowReportRecord, column: str) -> Union[str, None, int, float]:
+        """
+        Reads field value from row and transforms string type field to numeric if possible
+        """
+        value = row.value(column)
+        if value == "":
+            return None
+
+        if value is not None and column in REPORT_FIELD_TYPES:
+            if REPORT_FIELD_TYPES[column] == "integer":
+                value = 0 if value == "--" else int(value.replace(",", ""))
+            elif REPORT_FIELD_TYPES[column] == "number":
+                value = 0.0 if value == "--" else float(value.replace("%", "").replace(",", ""))
```

Comment:
Could this be incorrectly mutating values for percentages?
e.g. `float("27%".replace("%", ""))` becomes 27 when it should really be 0.27